contracts/beyondFramework/dca/base/DCABaseUpgradeable.sol

```diff
@@ -0,0 +1,1148 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import { InvestQueueLib } from "../libraries/InvestQueueLib.sol";
+import { DCAHistoryLib } from "../libraries/DCAHistoryLib.sol";
+import { IDCAStrategy } from "../interfaces/IDCAStrategy.sol";
+import { SwapLib } from "../libraries/SwapLib.sol";
+import { PortfolioAccessBaseUpgradeable } from "./PortfolioAccessBaseUpgradeable.sol";
+
+import { IERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/interfaces/IERC20Upgradeable.sol";
+import { SafeERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import { OwnableUpgradeable } from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import { PausableUpgradeable } from "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+import { ReentrancyGuardUpgradeable } from "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+
+abstract contract DCABaseUpgradeable is
+    PortfolioAccessBaseUpgradeable,
+    PausableUpgradeable,
+    ReentrancyGuardUpgradeable,
+    IDCAStrategy
+{
+    using InvestQueueLib for InvestQueueLib.InvestQueue;
+    using DCAHistoryLib for DCAHistoryLib.DCAHistory;
+    using SwapLib for SwapLib.Router;
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    DepositFee public depositFee;
```

Comment:
Leant from my own mistakes: I think it would be important to put all these state variable to a BaseStorageLibrary.sol and place them in a struct onto a specific slot (we could also got rid of _gap in that case). The reason is that when we break up a big DCAFancyStrategy into libraries, they many times need the variables from the base classes. One very easy and efficient way for those libraries to access those base variables could be done by reading that specific slot.

(This would have helped me a lot when I broke up the DNS strategy)